javascriptv3/example_code/bedrock-runtime/scenarios/converse_tool_scenario/converse-tool-scenario.js

```diff
@@ -0,0 +1,314 @@
+} from "@aws-sdk/client-bedrock-runtime";
+
+import { parseArgs } from "node:util";
+import { fileURLToPath } from "node:url";
+import { dirname } from "node:path";
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+import data from "./questions.json" with { type: "json" };
+import toolConfig from "./tool_config.json" with { type: "json" };
+
+const systemPrompt = [
+  {
+    text:
+      "You are a weather assistant that provides current weather data for user-specified locations using only\n" +
+      "the Weather_Tool, which expects latitude and longitude. Infer the coordinates from the location yourself.\n" +
+      "If the user provides coordinates, infer the approximate location and refer to it in your response.\n" +
+      "To use the tool, you strictly apply the provided tool specification.\n" +
+      "\n" +
+      "- Explain your step-by-step process, and give brief updates before each step.\n" +
+      "- Only use the Weather_Tool for data. Never guess or make up information. \n" +
+      "- Repeat the tool use for subsequent requests if necessary.\n" +
+      "- If the tool errors, apologize, explain weather is unavailable, and suggest other options.\n" +
+      "- Report temperatures in °C (°F) and wind in km/h (mph). Keep weather reports concise. Sparingly use\n" +
+      "  emojis where appropriate.\n" +
+      "- Only respond to weather queries. Remind off-topic users of your purpose. \n" +
+      "- Never claim to search online, access external data, or use tools besides Weather_Tool.\n" +
+      "- Complete the entire process until you have all required data before sending the complete response.",
+  },
+];
+const tools_config = toolConfig;
+
+/// Starts the conversation with the user and handles the interaction with Bedrock.
+async function askQuestion(userMessage) {
+  // The maximum number of recursive calls allowed in the tool use function.
+  // This helps prevent infinite loops and potential performance issues.
+  const max_recursions = 5;
+  const messages = [
+    {
+      role: "user",
+      content: [{ text: userMessage }],
+    },
+  ];
+  try {
+    const response = await SendConversationtoBedrock(messages);
+    await ProcessModelResponseAsync(response, messages, max_recursions);
+  } catch (error) {
+    console.log("error ", error);
+  }
+}
+
+// Sends the conversation, the system prompt, and the tool spec to Amazon Bedrock, and returns the response.
+// param "messages" - The conversation history including the next message to send.
+// return - The response from Amazon Bedrock.
+async function SendConversationtoBedrock(messages) {
+  const bedRockRuntimeClient = new BedrockRuntimeClient({
+    region: "us-east-1",
+  });
+  const modelId = "amazon.nova-lite-v1:0";
+  const response = await bedRockRuntimeClient.send(
+    new ConverseCommand({
+      modelId: modelId,
+      messages: messages,
+      system: systemPrompt,
+      toolConfig: tools_config,
```

Comment:
This action should handle a ModelNotReady exception and a generic BedrockRuntimeException.

Reply:
Done